lib/aggregate/aggregate_store.rb

```diff
@@ -76,7 +76,7 @@ def changed?
     end
 
     def set_changed
-      @changed = true
+      @changed = aggregate_values != aggregate_initial_values
       aggregate_owner&.set_changed
```

Comment:
I'm a little unclear about what the `aggregate_owner` is here. Should we only call `set_changed` on it if the values have changed?

Reply:
We technically only do if the values have changed.

I _think_ the `aggregate_owner` is when one Aggregate Object has an aggregate attribute (so, nested aggregate attributes)

We only call this `set_changed` if there _was_ a change, and we have to re-determine if there even was any change at all because it can go from (`true` -> `false`)

Reply:
Ah, good point that we only call this if there was a change. When I said "if the values have changed" I meant "if we've set @changed to true in the line before this".

I think what you're saying is this `set_changed` call will actually come back to this `AggregateStore#set_changed` method, if there is a nested aggregate attribute. In that case, this seems good to me. It'll go back into this method and determine if the `aggregate_owner` should have the `@changed` attribute set to true or false.

What I was worried about was whether this `set_changed` could be a callback that a class that is using aggregate could define. But it's not looking like that, so I think it's good the way it is.

Reply:
Ah! Yeah, it's just calling the same `set_changed` method to mark the "parent" as changed as well. Thanks for double checking that!